.gitignore

```diff
@@ -30,6 +30,11 @@ build/
 # spa_c_files builds
 pvlib/spa_c_files/build*
 
+# spa_c_files source
+pvlib/spa_c_files/spa.c
+pvlib/spa_c_files/spa.h
+pvlib/spa_c_files/spa_tester.c
+
```

Comment:
You are right, I see. Sorry that I was causing this confusion. The problem that the `so` will stay in the file structure exists, because it is ignored and will stay at its place, if you switch branches.